lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -872,10 +872,11 @@ public function walkIndexBy($indexBy)
      * Walks down a RangeVariableDeclaration AST node, thereby generating the appropriate SQL.
      *
      * @param AST\RangeVariableDeclaration $rangeVariableDeclaration
+     * @param bool $buildNestedJoins
      *
      * @return string
      */
-    public function walkRangeVariableDeclaration($rangeVariableDeclaration)
+    public function walkRangeVariableDeclaration($rangeVariableDeclaration, $buildNestedJoins = false)
```

Comment:
The additional parameter is problematic. The decision is one in the `SqlWalker` anyway, so the correct solution (IMO) would be to move the method body to a private method, then keep the public API as-is and delegate the call on the public API to the private API, where the new parameter is defined.